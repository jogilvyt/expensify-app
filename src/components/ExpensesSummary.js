import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal}) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">Add expense</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const filteredExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: filteredExpenses.length,
    expensesTotal: expensesTotal(filteredExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);