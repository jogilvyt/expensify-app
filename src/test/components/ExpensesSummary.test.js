import { shallow } from 'enzyme';
import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should show summary for 1 expense correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={expenses[0].amount} />);
  expect(wrapper).toMatchSnapshot();
});

test('should show summary for multiple expenses correctly', () => {
  const total = expenses[0].amount + expenses[1].amount + expenses[2].amount;
  const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={total} />);
  expect(wrapper).toMatchSnapshot();
});