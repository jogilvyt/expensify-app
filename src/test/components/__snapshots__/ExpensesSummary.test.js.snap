// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show summary for 1 expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $1.95
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should show summary for multiple expenses correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        $1,141.95
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;
